### DISCUSSION TITLE
Trying to install on Linux

### DISCUSSION THREAD

**Discussion by jdecent** [eliranwong/UniqueBible#1264] [Q&A]
I'm trying to install UBA on PopOS(based on Ubuntu). Running 'python3 uba.py' keeps telling me to install gdown. Running 'pip3 list' shows gdown installed.

**Comment by eliranwong** [comment]
can you try:

> pip3 install gdown

> cd UniqueBible

> touch use_system_site_packages

I experienced similar issue on Arch Linux.  I installed gdown globally and it works ... not sure if it is the same to you.

remarks: remove the venv_ ... folder and try try to start UBA again.

please kindly let me know if it works to you or not, thank you

**Reply by jdecent** [reply]
So I got past the gdown message after deleting the venv directory.  It installed a bunch of dependencies then stopped at "This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb."

**Comment by jdecent** [comment]
Running 'sudo apt-get install libxcb-xinerama0' made it work.

**Reply by otseng** [reply]
Added your findings to  https://github.com/eliranwong/UniqueBible/wiki/Installation-troubleshooting

**Comment by pvt02** [comment]
I posted the following in the wishlist section:

" Hello. I just found out about this program, and I am trying to figure out how to install it. Here is my request: please make an appimage of this program to make it easier to use. Appimages just work, and are the simplest to use on Linux; even common peons can use them right away. I am using an old bblx file of yours on a wine version of E-sword, which led me here. Thanks. "

--------

EDIT FEW HOURS LATER:
   Well I tried installing it with the download ZIP file instead of Git. Keep in mind I am not CLI warrior. I just am an ex-Windows user trying to do stuff, and reading all the "type this into Bash to make it work" tips ... Anyhow I typed in ALL the troubleshooting tips like the slightly above noob level monkey that I am, going down the page one at a time, and got to the part where it says, "export QT_DEBUG_PLUGINS=1
python3 main.py" hit enter and the program opened. I then messed with UniqueBible a bit, and then closed it. I then tried to open it with the desktop link. No go. I then tried to repeating the troubling/trouble shooting steps. No go. I tried all the troubleshooting steps above on this page. No go. I tried all sorts of stuff outside the  UniqueBible Git, and messed up Cinnamon somehow (I use Devuan Linux). I reinstalled Cinnamon, and then deleted the UniqueBible folder, and tried all the steps from before. NO GO. Oh well, like I said before an Appimage would be better, or at least a regular deb file that Devuan can install automatically.

The HSB2.bblx file I referenced above, as well as the interlinears in Unique Bible have at least the following errors. First a wau prefix (meaning and) is not always in the correct translated location, and someone with no knowledge of Hebrew would thus be mislead. Also when "of" is translated as in the Spirit of or Face of, the of's that I saw were also put in the wrong place as they belong to the first word not the second so for example Face-OF the deep not Face of-the-deep. The genitive belongs to the first word, not the second word as in Latin. Other than such things I do find the file useful, which is why I am here trying to get UniqueBible to work to see if it is even more useful.

Moreover, trying all the CLI and so forth gymnastics to try and get it to work messed up my Wine installation (why?), and God knows what else, so now I can not use E-sword, or any of my Wine programs. Good thing I made a clone of my drive recently, as that is the easy way to get my programs working again. However AGAIN if we had an Appimage of the UniqueBible all this would be a non-issue.

**Reply by otseng** [reply]
Yeah, it's not very easy to get UBA running, on any platform.  AppImage looks like a great idea.  I'll explore it and hopefully it can work for UBA.

**Reply by pvt02** [reply]
thanks
